src/vs/workbench/parts/extensions/browser/extensionsActions.ts

```diff
@@ -1133,6 +1134,120 @@ export class ShowWorkspaceRecommendedExtensionsAction extends Action {
 	}
 }
 
+export class InstallWorkspaceRecommendedExtensionsAction extends Action {
+
+	static ID = 'workbench.extensions.action.installWorkspaceRecommendedExtensions';
+	static LABEL = localize('installWorkspaceRecommendedExtensions', "Install All Workspace Recommended Extensions");
+
+	private disposables: IDisposable[] = [];
+	private recommendations: string[] = [];
+	private toInstall: string[] = [];
+	private updatePromise: TPromise<any> = TPromise.as(null);
+
+	constructor(
+		id: string = InstallWorkspaceRecommendedExtensionsAction.ID,
+		label: string = InstallWorkspaceRecommendedExtensionsAction.LABEL,
+		@IWorkspaceContextService private contextService: IWorkspaceContextService,
+		@IViewletService private viewletService: IViewletService,
+		@IExtensionsWorkbenchService private extensionsWorkbenchService: IExtensionsWorkbenchService,
+		@IExtensionTipsService private extensionTipsService: IExtensionTipsService,
+		@IMessageService private messageService: IMessageService
+	) {
+		super(id, label, 'extension-action');
+		this.extensionsWorkbenchService.onChange(() => this.updateToInstall(), this, this.disposables);
+		this.contextService.onDidChangeWorkbenchState(() => this.updateRecommendations(), this, this.disposables);
+		this.updatePromise = this.updateRecommendations();
```

Comment:
@ramya-rao-a I would not store the state of workspace recommendations here. I would always ask `IExtensionsTip` service for the workspace recommendations while running. This would simplify the action a lot and I do not see the advantage of maintaining the state here except causing more maintenance.

Reply:
Calling `IExtensionTipService` to get workspace recommendations involves reading and parsing the `extensions.json` file.
Doing this for every change in extensionWorkbenchService felt redundant which is why I chose to store the recommendations.

I am ok either way. Made the change as requested.

Reply:
Then, I would cache it in the service that owns recommendations and not in callers .